src/Init/Data/Char/UnicodeSkipList.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean-Baptiste Tristan
+-/
+prelude
+import Init.Data.Range
+
+/-!
+The unicode skip list data structure is used to represent the set of characters
+that have some property. The data structure is precomputed using the
+tool in script/unicode.
+
+We use a variant of the skip list data structure that takes advantage
+of the distribution of character properties for the common properties of
+interest. Considering the sequence of codepoints that satisfy a given
+property, they tend to appear is small ranges that are clustered. These
+clusters are separated from each other by large ranges of codepoints that
+do not satisfy the property. We refer to such clusters as runs.
+
+The data structure takes advantage of this by representing the set
+using an array of 32 bit integers (the outer array that allows to skip)
+and an array of 8 bit integers. The integers of the outer array
+contains enough information to let us locate in which run a codepoint
+belong, and we call it `runs`. The bytes of the inner array, which we call
+`offsets` record the length of the ranges. Large ranges of codepoints
```

Comment:
Nothing in the text here explains that the numbers in `offsets` are alternately the lengths of sequences satisfying the property and lengths of sequences not satisfying the property: this is only apparent when we get to the example.

This also makes the reference to "parity" in the next paragraph impossible to interpret until later.